edx/edx-arch-experiments
# feat: Add Waffle config watcher

Listens to Waffle model changes and reports them 1) to the log and 2) to a Slack webhook.

This adds a dependency on django-waffle.

----

To test this in devstack:

- Check out the branch under src, e.g. `git worktree add ../src/edxarch-waffle/ timmc/waffle-watcher`
- In lms-shell: `pip install -e /edx/src/edxarch-waffle/`
- In edx-platform's lms/envs/common.py, add `'edx_arch_experiments.config_watcher'` to INSTALLED_APPS
- Log in to LMS as the edx user
- Change, add, or delete a Waffle flag, switch, or sample
- Watch for the log message

You can also configure `CONFIG_WATCHER_SLACK_WEBHOOK_URL` to a request bin to confirm the request format.

----

**Merge checklist:**
Check off if complete *or* not applicable:
